lib/SPIRV/libSPIRV/SPIRVFunction.cpp

```diff
@@ -150,7 +150,10 @@ SPIRVFunction::decodeBB(SPIRVDecoder &Decoder) {
 
     SPIRVInstruction *Inst = static_cast<SPIRVInstruction *>(Decoder.getEntry());
     assert(Inst);
-    BB->addInstruction(Inst);
+    if (Inst->getOpCode() == OpUndef)
+      Module->addUndef(Inst->getType());
```

Comment:
If two functions define Undef of the same type, we will have these two Undef instructions in module scope along with other constants. They will be of the same type, but with different ids.
See also #169